core/vm/runtime/impl/runtime_impl.cpp

```diff
@@ -110,11 +112,50 @@ namespace fc::vm::runtime {
     return fc::outcome::success();
   }
 
-  fc::outcome::result<void> RuntimeImpl::deleteActor() {
-    // TODO: transfer to kBurntFundsActorAddress
-    // TODO(a.chernyshov) FIL-137 implement state_tree remove if needed
-    // return state_tree_->remove(address);
-    return fc::outcome::failure(RuntimeError::kUnknown);
+  fc::outcome::result<void> RuntimeImpl::deleteActor(const Address &address) {
+    OUTCOME_TRY(chargeGas(execution_->env->pricelist.onDeleteActor()));
+    auto &state{*execution()->state_tree};
+    if (auto _actor{state.get(getCurrentReceiver())}) {
+      if (_actor.has_error() && _actor.error() == HamtError::kNotFound) {
```

Comment:
`if (auto _actor{error})` will go to `else`, so `if (_actor.has_error() && ...)` is always false. Move to `else` or remove

Reply:
fixed